guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/RangeDeserializer.java

```diff
@@ -193,6 +216,42 @@ public Range<?> deserialize(JsonParser p, DeserializationContext context)
         }
     }
 
+    private Range<?> deserializeRangeFromString(DeserializationContext context, JsonParser p)
+            throws IOException
+    {
+        String rangeInterval = p.getText();
+
+        if (rangeInterval == null || rangeInterval.isEmpty())
+        {
+            return null;
+        }
+
+        BoundType lowerBoundType = rangeInterval.startsWith("[") ? BoundType.CLOSED : BoundType.OPEN;
+        BoundType upperBoundType = rangeInterval.endsWith("]") ? BoundType.CLOSED : BoundType.OPEN;
+
+        rangeInterval = rangeInterval.substring(1, rangeInterval.length() - 1);
+        String[] parts = rangeInterval.split("\\.\\.");
+
+        if (parts.length == 2)
+        {
+            boolean isLowerInfinite = parts[0].equals("-∞");
+            boolean isUpperInfinite = parts[1].equals("+∞");
+
+            if (isLowerInfinite && isUpperInfinite) {
+                return RangeFactory.all();
+            } else if (isLowerInfinite) {
+                return RangeFactory.upTo(Integer.parseInt(parts[1]), upperBoundType);
+            } else if (isUpperInfinite) {
+                return RangeFactory.downTo(Integer.parseInt(parts[0]), lowerBoundType);
+            } else {
+                return RangeFactory.range(Integer.parseInt(parts[0]), lowerBoundType,
+                        Integer.parseInt(parts[1]), upperBoundType);
+            }
+        }
+
+        return null;
```

Comment:
I think this should rather fail with message indicating invalid range description, using `DeserializationContext.handleWeirdStringValue()`. Example from this same class:

        return (BoundType) context.handleWeirdStringValue(BoundType.class, name,
                "not a valid BoundType name (should be one of: %s)",
                Arrays.asList(BoundType.values()));